File: content/problems/1974-minimum-time-to-type-word-using-special-typewriter/solutions/circular-letter-distance/Solution.kt
class Solution {
    fun minTimeToType(word:String):Int{
        var answer=0;
        var current='a';
        for(target in word){
            val difference=kotlin.math.abs(target-current);
            answer+=minOf(difference,26-difference)+1;
            current=target
        };
        return answer
    }
}

File: content/problems/1974-minimum-time-to-type-word-using-special-typewriter/solutions/circular-letter-distance/solution.cpp
class Solution {
    public: int minTimeToType(string word){
        int answer=0;
        char current='a';
        for(char target:word){
            int difference=abs(target-current);
            answer+=min(difference,26-difference)+1;
            current=target;
        }
        return answer;
    }
};
